src/k8s/api/v1/cluster.go

```diff
@@ -22,6 +22,24 @@ type GetKubeConfigResponse struct {
 	KubeConfig string `json:"kubeconfig"`
 }
 
+// JoinClusterRequest is used to request the configuration to join the k8s cluster.
+type JoinClusterRequest struct {
+	Token string `json:"token"`
+}
+
+// JoinClusterResponse is the response for "POST 1.0/k8sd/cluster/join".
+type JoinClusterResponse struct {
+	// TODO: use named arguments (e.g. ExtraKubeletArgs) instead.
+	ExtraServiceArgs ExtraServiceArgs `json:"extraServiceArgs"`
```

Comment:
Note that this is not just a matter of TODO here, since it might be hard to retrofit without an API change. Not a problem (and there are ways to embed structs and hopefully preserve backwards-compatibility in the future.

Please also add a godoc here, explaining how ExtraServiceArgs are used.